crates/compression-codecs/src/zstd/decoder.rs

```diff
@@ -84,3 +86,14 @@ impl Decode for ZstdDecoder {
         Ok(bytes_left == 0)
     }
 }
+
+impl DecodedSize for ZstdDecoder {
+    fn decoded_size(input: &[u8]) -> Result<u64> {
+        zstd_safe::find_frame_compressed_size(input)
+            .map_err(|error_code| io::Error::other(get_error_name(error_code)))
+            .and_then(|size| {
+                size.try_into()
+                    .map_err(|_| io::Error::from(io::ErrorKind::InvalidData))
```

Comment:
I think [FileTooLarge](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.FileTooLarge) might be better?

Or we can have `Error::other(format!("... {err}"))`